backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ClientProviderImpl.java

```diff
@@ -19,30 +19,101 @@
 package org.apache.james.backends.es;
 
 import java.net.InetAddress;
-import java.net.UnknownHostException;
+import java.util.Objects;
 
 import org.elasticsearch.client.Client;
 import org.elasticsearch.client.transport.TransportClient;
 import org.elasticsearch.common.transport.InetSocketTransportAddress;
 
-import com.google.common.base.Throwables;
+import com.github.fge.lambdas.Throwing;
+import com.github.fge.lambdas.consumers.ConsumerChainer;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
 
 public class ClientProviderImpl implements ClientProvider {
 
-    private final String host;
-    private final int port;
-    
-    public ClientProviderImpl(String host, int port) {
-        this.host = host;
-        this.port = port;
+    public static class Host {
+        private final String host;
+        private final int port;
+
+        public Host(String host, int port) {
```

Comment:
It's a bit redondant with CassandraNodeIpAndPort class maybe we should made a generic IpAndPort class